webapp/src/ts/actions/reports.ts

```diff
@@ -86,16 +91,16 @@ export class ReportsActions {
   }
 
   clearSelection() {
+    this.store.dispatch(Actions.setSelectedReport(undefined));
     this.store.dispatch(Actions.setSelectedReports([]));
-    // setVerifyingReport(false);
   }
 
-  setSelectMode(value) {
-    return this.store.dispatch(Actions.setSelectMode(value));
+  setSelectMode() {
+    return this.store.dispatch(Actions.setSelectMode());
```

Comment:
Not sure about the definition of `toggle`, to me means "a switch of a binary state" so anything that switches it will be a toggle? 😅 

The value is a "calculation" based on the previous value of `selectMode` and the current state of `selectedReports`. 
I'm not big fan of effects, so I tried to avoid it but it was getting too messy and at the end it made more sense (to me) to centralise this logic in an effect.

Reply:
Yes, I think toggles are confusing because you don't know what state your action will end up switching to, because of delays or other interruption. 
Picture this: 
- you have a toggle to set something on or off
- you update the state for the client to see that the state is on or off
- client starts clicking on-off and has slow internet or the quick actions produce lag
- the client will end up seeing the state is on, and clicks the button to be off but in the end the state is still on, because that by the time the request makes it to the "server", the state had already been set to "off" by another previous action. 

It will be confusing if the client explicitly sets the state on or off, instead of relying on the action to reach the server with the correct state.

Reply:
Okay, I've moved the logic to set the select mode to reports component and it's based on the store data, by doing that we don't need anymore the effect and the report's action to set it.  
I think it's simpler and easier to understand :)